### PR TITLE
Feature: Added BaseDataForm component to aggregate all required data

### PR DESCRIPTION
**Pull request by devtobi**
# Pull Request

## Changes

- Added BaseDataForm component for additional fields (name, committeeSize) and later implemented unions
- Added aggregated validation in BaseDataForm component
- Organized components into subfolders
- Added validation trigger in GroupDataTable when committeeSize was changed from the outside
- Moved calculation of group statistics (num of rows, total sums of seats and votes) into composable as its reused in both GroupDataTableSummaryRow and BaseDataForm

## Checklist

**Note**: If some checklist items are not relevant for your PR, just remove them.

- [x] Met all acceptance criteria of the issue
- [x] Added meaningful PR title and list of changes in the description
- [x] Wrote code and comments in English

### CODE REVIEW

File: sitzverteilung-frontend/src/components/basedata/groupdata/GroupDataTable.vue [hunk 1]
```diff
@@ -109,11 +109,11 @@ import type { Group } from "@/types/Group";
 
 import { mdiAccountGroup, mdiDelete, mdiSeat, mdiVote } from "@mdi/js";
 import { useDebounceFn, useTemplateRefsList } from "@vueuse/core";
-import { computed, ref, useTemplateRef } from "vue";
+import { computed, ref, useTemplateRef, watch } from "vue";
```
[the remark this answers is not in the document]
Comment (devtobi) [reply sitzverteilung-frontend/src/components/basedata/groupdata/GroupDataTable.vue:113]: Thats not true, vue also supports useTemplateRef

File: sitzverteilung-frontend/src/composables/useGroupStatistics.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+export const useGroupStatistics = (
+  groups: Ref<Group[]>,
+  expectedSeats: Ref<number>
+) => {
+  const amountOfGroups = computed(() => groups.value.length);
+  const isTooManyGroups = computed(
+    () => amountOfGroups.value > expectedSeats.value
+  );
+  const isSeatsTooHigh = computed(() => totalSeats.value > expectedSeats.value);
```
Comment (GerhardPx) [inline sitzverteilung-frontend/src/composables/useGroupStatistics.ts:14]: We could combine these two properties into a combined computed property for validation, especially because they are mutually exclusive.

File: sitzverteilung-frontend/src/components/basedata/BaseDataForm.vue [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+function checkCommitteeSizeField(event: KeyboardEvent) {
+  preventNonNumericInput(event);
+  preventTooLongInput(
+    baseData.value.committeeSize?.toString() ?? "",
+    maxCommitteeSizeLength.value,
+    event
+  );
+}
+
+const seatFieldValidationError = computed(() => {
```
Comment (GerhardPx) [inline sitzverteilung-frontend/src/components/basedata/BaseDataForm.vue:98]: Right now, there are no tests, I think. We should at least add tests for the validation of the committeeSize.

File: sitzverteilung-frontend/src/composables/useGroupStatistics.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import type { Group } from "@/types/Group";
+import type { Ref } from "vue";
+
+import { computed } from "vue";
+
+export const useGroupStatistics = (
```
Comment (GerhardPx) [inline sitzverteilung-frontend/src/composables/useGroupStatistics.ts:6]: We should also consider adding tests for useGroupStatistics, to see the correct calculation of seats and how it handles an empty array.

File: sitzverteilung-frontend/src/components/basedata/BaseDataForm.vue [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  ...baseDataNames,
+  baseData.value.name,
+]);
+
+const expectedSeats = computed(() =>
+  !(baseData.value.committeeSize === 0 || !!baseData.value.committeeSize)
+    ? 0
+    : baseData.value.committeeSize
+);
+const { isTooManyGroups, isSeatsTooLow, isSeatsTooHigh } = useGroupStatistics(
```
Comment (GerhardPx) [inline sitzverteilung-frontend/src/components/basedata/BaseDataForm.vue:118]: Tests for the correct display of useGroupStatistics can't hurt either.

File: sitzverteilung-frontend/src/composables/useGroupStatistics.ts (hunk 2 above)
Comment (devtobi) [reply sitzverteilung-frontend/src/composables/useGroupStatistics.ts:14]: Combining would loose the ability to show 2 different error messages (either not enough seats or too many seats). I don't think that would be good.

File: sitzverteilung-frontend/src/components/basedata/BaseDataForm.vue (hunk 3 above)
Comment (devtobi) [reply sitzverteilung-frontend/src/components/basedata/BaseDataForm.vue:98]: So you suggest to test the `useGroupStatistics` composable in this case? The `seatFieldValidationError` just translates occuring validation problems to custom error messages where it can not be evaluated on a single field level.

File: sitzverteilung-frontend/src/composables/useGroupStatistics.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      0
+    );
+  });
+
+  const totalVotes = computed(() => {
+    return groups.value.reduce((sum, group) => sum + (group.votes ?? 0), 0);
+  });
+
+  return {
+    amountOfGroups,
```
[the remark this answers is not in the document]
Comment (GerhardPx) [reply sitzverteilung-frontend/src/composables/useGroupStatistics.ts:35]: Making an interface with these properties, although you are right, they are automatically inferred here

**Review by GerhardPx — APPROVED** [review]
